Decouple Desktop UI into monorepo app

webfiltered opened the pull request:
## Summary

Extracts desktop UI into apps/desktop-ui package with minimal changes.

## Changes

- **What**:
  - Separates desktop-specific code into standalone package with independent Vite config, router, and i18n
  - Drastically simplifies the main app router by removing all desktop routes
  - Adds a some code duplication, most due to the existing design
  - Some duplication can be refactored to be *simpler* on either side - no need to split things by `isElectron()`
  - Rudimentary storybook support has been added
- **Breaking**: Stacked PR for publishing must be merged before this PR makes it to stable core (but publishing _could_ be done manually)
  - #5915
- **Dependencies**: Takes full advantage of pnpm catalog. No additional dependencies added.

## Review Focus

- Should be no changes to normal frontend operation
- Scripts added to root package.json are acceptable
- The duplication in this PR is copied as is, wherever possible. Any corrections or fix-ups beyond the scope of simply migrating the functionality as-is, can be addressed in later PRs.  That said, if any changes are made, it instantly becomes more difficult to separate the duplicated code out into a shared utility.
  - Tracking issue to address concerns: #5925

### i18n

Fixing i18n is out of scope for this PR.  It is a larger task that we should consider carefully and implement properly.  Attempting to isolate the desktop i18n and duplicate the _current_ localisation scripts would be wasted energy.

webfiltered committed:
Add app scaffolding (Vite, TS, router, i18n)

Adds apps/desktop-ui package setup, Vite config, TypeScript config, base index.html, public assets, env/types, and base router/i18n/main wiring.

webfiltered committed:
Move desktop code to apps/desktop-ui (pure moves)

Relocates Desktop UI components, views, stores, types, and constants under apps/desktop-ui with no content changes (pure moves).

webfiltered committed:
Adjust root src/router.ts for desktop app split

Updates existing app router to accommodate Desktop UI extraction.

webfiltered committed:
Add duplicate components / utils

webfiltered committed:
Adjust CODEOWNERS

webfiltered committed:
Update package lock

webfiltered committed:
Fix desktop-ui import paths and remove broken stories file

webfiltered committed:
Configure desktop-ui dev server to use port 5174

webfiltered committed:
Fix compiler errors in vite config

webfiltered committed:
Add root repo scripts for desktop-ui dev and build

webfiltered committed:
Decouple desktop-ui from root vite config

webfiltered committed:
Add shared-frontend-utils dependency

webfiltered committed:
Use shared-frontend-utils package in desktop-ui

webfiltered committed:
Remove private flag from desktop-ui

Added by pattern from packages/* packages.

webfiltered committed:
[ts] Fix type issue

webfiltered committed:
Autoformat

webfiltered committed:
Add missing dependencies to desktop-ui

webfiltered committed:
Remove redundant ts `@` path alias

webfiltered committed:
Update package lock file

webfiltered committed:
Rename root #vue-app HTML ID to prevent conflict

Copies style to local style.css

webfiltered committed:
Fix i18n collection script broken by migration

webfiltered committed:
Remove unused files; migrated to desktop app

webfiltered committed:
Remove unused function

webfiltered committed:
Remove unused const - only used in installer

webfiltered committed:
Remove unused function (i18n d)

webfiltered committed:
Remove unused code

webfiltered committed:
Remove unused export

webfiltered committed:
Restore desktop InstallView stories

webfiltered committed:
Remove unused code - should not have been copied

webfiltered committed:
Isolate desktop stories to fix @ alias clash

webfiltered committed:
Allow serving desktop Storybook via Nx

webfiltered committed:
Add desktop storybook script

christian-byrne reviewed (dismissed):
LGTM

christian-byrne commented:
> @christian-byrne Did you guys get codex working on the repo? Can it review?

@benceruleanlu can you remind me how it's done again?

DrJKL commented on scripts/collect-i18n-general.ts:
```diff
@@ -1,12 +1,12 @@
 import * as fs from 'fs'
 
+import { DESKTOP_DIALOGS } from '../apps/desktop-ui/src/constants/desktopDialogs'
```
Want to add a path in the root tsconfig?
Maybe `'@desktop/*': './apps/desktop-ui/src'`

DrJKL commented on apps/desktop-ui/src/assets/css/style.css:
```diff
@@ -0,0 +1,6 @@
+@import '@comfyorg/design-system/css/style.css';
+
+#desktop-app {
+  position: absolute;
+  inset: 0;
+}
```
Can't use tailwind utilities for these?

DrJKL commented on apps/desktop-ui/src/components/bottomPanel/tabs/terminal/BaseTerminal.vue:
```diff
@@ -0,0 +1,113 @@
+      size="small"
+      :class="
+        cn('absolute top-2 right-8 transition-opacity', {
+          'opacity-0 pointer-events-none select-none': !isHovered
```
The CSS hover doesn't work for this?

DrJKL commented on apps/desktop-ui/src/components/bottomPanel/tabs/terminal/BaseTerminal.vue:
```diff
@@ -0,0 +1,113 @@
+  created: [ReturnType<typeof useTerminal>, Ref<HTMLElement | undefined>]
+  unmounted: []
+}>()
+const terminalEl = ref<HTMLElement | undefined>()
+const rootEl = ref<HTMLElement | undefined>()
```
nit: The possibly-undefinedness is implicit when you don't pass in a default.
```suggestion
const terminalEl = ref<HTMLElement>()
const rootEl = ref<HTMLElement>()
```

DrJKL commented on apps/desktop-ui/src/main.ts:
```diff
@@ -0,0 +1,46 @@
+import { i18n } from './i18n'
+import router from './router'
+
+const ComfyUIPreset = definePreset(Aura, {
+  semantic: {
+    // @ts-expect-error fixme ts strict error
+    primary: Aura['primitive'].blue
+  }
+})
```
Should some of this also be moved to the design system package?

DrJKL commented on apps/desktop-ui/tsconfig.json:
```diff
@@ -0,0 +1,24 @@
+  },
+  "include": [
+    ".storybook/**/*",
+    "src/**/*.ts",
+    "src/**/*.mts",
+    "src/**/*.cts",
+    "src/**/*.tsx",
+    "src/**/*.vue",
+    "src/**/*.json",
+    "src/**/*.d.ts",
```
Are all of these used?

DrJKL replied:
(You could also probably unify these with `{ts,mts,cts...}`

webfiltered committed:
Remove unnecessary globs from tsconfig

webfiltered reviewed (commented):
The original images were not removed, and are being duplicated here for likely no reason.  Comfy brand mark will probably see re-use.  This can be moved to an assets package.

webfiltered committed:
Remove migrated Desktop UI images

webfiltered committed:
Restore storybook config to migration of base repo

Originally LLM-parsed and "fixed"

webfiltered committed:
Fix Storybook won't build

webfiltered committed:
Fix Storybook build by removing broken external config

webfiltered committed:
Remove broken external config from root Storybook

The external: () => false configuration causes a PostCSS error
during CSS URL processing. Removing it fixes the build without
affecting dependency bundling.
